lib/Url/UrlManager.php

```diff
@@ -233,13 +233,25 @@ public function isUserPath()
 
     /**
      * @param int $profileId
+     * @param int $articleId
      *
      * @throws \rex_sql_exception
      *
      * @return null|UrlManager[]
      */
-    public static function getByProfileId($profileId)
+    public static function getByProfileId($profileId, $articleId)
     {
+        $Article = \rex_article::get($articleId);
+        $artPath = $Article->getPathAsArray();
+
+        foreach ($artPath as $pathId) {
+            $Category = \rex_category::get($pathId);
+
+            if (!$Category->isOnline()) {
+                return null;
+            }
+        }
+
```

Comment:
Diese Abfrage gehört für mich weniger in das Url-Addon, da es auch so nicht REDAXO selbst vorgibt. Jeder Artikel ist erst einmal aufrufbar. Will man Artikel oder hier Datensätze nicht verlinken müsste das eher im Projekt selbst abgehandelt werden. Mit dieser Schleife erhalte ich keine Urls sobald eine Kategorie offline ist.